tests/E2E Tests/WebAppUiTests/WebAppCallsApiCallsGraphLocally.cs

```diff
@@ -172,6 +157,125 @@ public async Task ChallengeUser_MicrosoftIdFlow_LocalApp_ValidEmailPasswordCreds
+            {
+                // Start the web app and api processes.
+                // The delay before starting client prevents transient devbox issue where the client fails to load the first time after rebuilding.
+                serviceProcess = UiTestHelpers.StartProcessLocally(_testAssemblyLocation, _devAppPathCiam + TC.s_myWebApiPath, TC.s_myWebApiExe, serviceEnvVars);
+                await Task.Delay(3000);
+                clientProcess = UiTestHelpers.StartProcessLocally(_testAssemblyLocation, _devAppPathCiam + TC.s_myWebAppPath, TC.s_myWebAppExe, clientEnvVars);
+
+                if (!UiTestHelpers.ProcessesAreAlive(new List<Process>() { clientProcess, serviceProcess }))
+                {
+                    Assert.Fail(TC.WebAppCrashedString);
+                }
+
+                var page = await NavigateToWebApp(context, WebAppCiamPort);
+
+                // Initial sign in
+                _output.WriteLine("Starting web app sign-in flow.");
+                string email = "idlab@msidlabciam6.onmicrosoft.com";
+                await UiTestHelpers.FirstLogin_MicrosoftIdFlow_ValidEmailPassword(page, email, LabUserHelper.FetchUserPassword("msidlabciam6"), _output);
+                await Assertions.Expect(page.GetByText("Welcome")).ToBeVisibleAsync(_assertVisibleOptions);
+                await Assertions.Expect(page.GetByText(email)).ToBeVisibleAsync(_assertVisibleOptions);
+                _output.WriteLine("Web app sign-in flow successful.");
+
+                // Sign out
+                _output.WriteLine("Starting web app sign-out flow.");
+                await page.GetByRole(AriaRole.Link, new() { Name = "Sign out" }).ClickAsync();
+                await UiTestHelpers.PerformSignOut_MicrosoftIdFlow(page, email, TC.LocalhostUrl + WebAppCiamPort + SignOutPageUriPath, _output);
+                _output.WriteLine("Web app sign out successful.");
+
+                // Sign in again using Todo List button
+                _output.WriteLine("Starting web app sign-in flow using sign in button after sign out.");
+                await page.GetByRole(AriaRole.Link, new() { Name = "Sign in" }).ClickAsync();
+                await UiTestHelpers.FirstLogin_MicrosoftIdFlow_ValidEmailPassword(page, email, LabUserHelper.FetchUserPassword("msidlabciam6"), _output);
+                await Assertions.Expect(page.GetByText("Welcome")).ToBeVisibleAsync(_assertVisibleOptions);
+                await Assertions.Expect(page.GetByText(email)).ToBeVisibleAsync(_assertVisibleOptions);
+                _output.WriteLine("Web app sign-in flow successful using Sign in button after sign out.");
+            }
+            catch (Exception ex)
+            {
+                Assert.Fail($"the UI automation failed: {ex} output: {ex.Message}.");
+            }
+            finally
+            {
+                // Add the following to make sure all processes and their children are stopped.
+                Queue<Process> processes = new();
```

Comment:
Can't this code be consolidated with other tests?

Reply:
I was thinking about it but there are quite a few variables in this code block that would need to be passed into the consolidated method that it would end up being more complicated than its worth. I will take another look at it though.